src/components/singleProductCard/SingleProductCard.jsx

```diff
@@ -0,0 +1,149 @@
+import React from "react";
+import "./SingleProductcard.css";
+import { BiRupee } from "react-icons/bi";
+import { FaTag, FaRegHeart, FaStar, FaHeart } from "react-icons/fa";
+import { Link } from "react-router-dom";
+import { useCartandWishList } from "../../context/CartAndWishlist-context";
+import toast from "react-hot-toast";
+
+export const SingleProductCard = ({ Product }) => {
+  const { cartState, cartDispatch, wishListState, wishListDispatch } =
+    useCartandWishList();
+
+  const { cartItem } = cartState;
+  const { wishListItem } = wishListState;
+  console.log(Product, "singlecard");
+  const {
+    image,
+    productName,
+    sellingPrice,
+    originalPrice,
+    discountOffer,
+    isRatingGood,
+    rating,
+    fastDelivery,
+    brand,
+    quantity,
+    description,
+    isSoldOut,
+  } = Product;
+  console.log(image);
+
+  const showToast = () => {
+    toast("inform you when product back in stock", { icon: "✔️" });
+  };
+
+  return (
+    <div className="product-image-details-container">
+      <div className="image-of-product mt-3">
+        {/* <span> */}
+
+        <img src={image} className="image-of-product" alt={productName} />
+        {/* </span> */}
+      </div>
+      <div className="product-details">
+        <h2 className="heading heading-of-single-product-page">
+          {productName}
+        </h2>
+        <div className="price-detail-container flex-center">
+          <span className="discount flex-center">
+            <BiRupee />
+            {sellingPrice}
+          </span>
+          <p className="price flex-center">
+            <BiRupee />
+            {originalPrice}
+          </p>
+          <p className="discount-tag-for-single-product text-size-md">
+            ({discountOffer})
+          </p>
+          <span className="rating-badge rating-badge-single-product">
+            <div
+              className={
+                isRatingGood
```

Comment:
I like that you are using different rating styles on a basis of good rating or not